build/builder.go

```diff
@@ -0,0 +1,24 @@
+package build
+
+import (
+	"github.com/fossas/fossa-cli/config"
+	"github.com/fossas/fossa-cli/module"
+)
+
+// A Builder is an implementation of functionality for different build systems.
+type Builder interface {
+	// Initialize gathers build environment information and does build setup.
+	Initialize() error
+	// Build runs a best-effort attempt at building the module.
+	Build(m module.Module, force bool) error
+	// Analyze returns the dependencies of a module.
+	Analyze(m module.Module, allowUnresolved bool) ([]module.Dependency, error)
+
+	// IsBuilt checks whether a module has been built.
+	IsBuilt(m module.Module, allowUnresolved bool) (bool, error)
+
+	// IsModule takes a configKey and returns whether it is valid to elect this Builder.
+	IsModule(configKey string) (bool, error)
```

Comment:
If you're not planning to use `IsModule` in `init`, remove it from the interface and remove all the implementations.

Reply:
I have plans for IsModule in `init`.